server/src/jobs/fiabilisation/uai-siret/apply-fiabilisation/index.js

```diff
@@ -0,0 +1,126 @@
+    jobname: JOB_NAME,
+    date: new Date(),
+    action: "beginning",
+  });
+
+  await updateDossiersApprenantAndOrganismesAFiabiliser();
+  await updateOrganismesFiabilisationStatut(FIABILISATION_TYPES.DEJA_FIABLE);
+  await updateOrganismesFiabilisationStatut(FIABILISATION_TYPES.NON_FIABILISABLE);
+};
+
+/**
+ * Méthode de maj des dossiersApprenants pour prise en compte de la fiabilisation UAI SIRET
+ */
+const updateDossiersApprenantAndOrganismesAFiabiliser = async () => {
+  const allCouplesAFiabiliser = await fiabilisationUaiSiretDb()
+    .find({ type: FIABILISATION_TYPES.A_FIABILISER })
+    .toArray();
+
+  let dossiersApprenantAfiabiliserModifiedCount = 0;
+  let organismesAFiabiliserModifiedCount = 0;
+
+  for (const fiabilisation of allCouplesAFiabiliser) {
+    try {
+      const { modifiedCount: dossiersApprenantModifiedCount } = await dossiersApprenantsMigrationDb().updateMany(
+        { ...filters, uai_etablissement: fiabilisation.uai, siret_etablissement: fiabilisation.siret },
+        {
+          $set: {
+            uai_etablissement: fiabilisation.uai_fiable,
+            siret_etablissement: fiabilisation.siret_fiable,
+          },
+        }
+      );
+      dossiersApprenantAfiabiliserModifiedCount += dossiersApprenantModifiedCount;
+
+      const { modifiedCount: organismesModifiedCount } = await organismesDb().updateMany(
+        { uai: fiabilisation.uai, siret: fiabilisation.siret },
+        { $set: { fiabilisation_statut: FIABILISATION_TYPES.FIABILISE } }
+      );
+      organismesAFiabiliserModifiedCount += organismesModifiedCount;
+    } catch (err) {
+      // TODO Fixer l'erreur de duplicate key sur l'id_erp_apprenant
+      logger.error(err);
+      await createJobEvent({
+        jobname: JOB_NAME,
+        date: new Date(),
+        action: "updateDossiersApprenantAndOrganismesAFiabiliser",
+        data: {
+          uai: fiabilisation.uai,
+          uai_fiable: fiabilisation.uai_fiable,
+          siret: fiabilisation.siret,
+          siret_fiable: fiabilisation.siret_fiable,
+          err,
+        },
+      });
+    }
+  }
+
+  await createJobEvent({
+    jobname: JOB_NAME,
+    date: new Date(),
+    action: "updateDossiersApprenantAndOrganismesAFiabiliser",
+    data: {
+      dossiersApprenantAfiabiliserModifiedCount,
+      organismesAFiabiliserModifiedCount,
+    },
+  });
+
+  logger.info(dossiersApprenantAfiabiliserModifiedCount, "dossiers apprenants a fiabiliser mis à jour");
+  logger.info(organismesAFiabiliserModifiedCount, "organismes a fiabiliser  mis à jour");
+};
+
+/**
+ *
+ * @param {*} fiabilisationType
+ */
+const updateOrganismesFiabilisationStatut = async (fiabilisationType) => {
+  const allFiabilisationCouples = await fiabilisationUaiSiretDb().find({ type: fiabilisationType }).toArray();
+
+  let organismesUpdatedCount = 0;
+
+  for (const fiabilisation of allFiabilisationCouples) {
+    try {
+      const { modifiedCount } = await organismesDb().updateMany(
```

Comment:
si bcp d'elements, ca risque d'etre un peu long. Sinon, tu peux faire un $lookup depuis fiabilisationUaiSiretDb sur la collection organismesDb, et faire un organismesDb.updateMany avec.
up to 2

Reply:
oui, j'ai repris l'ancien script de stephane sans vraiment me pencher sur l'optimisation.
Je te propose de mettre en todo la partie optim des requetes pour deploy une première version.